metrics/memory.go

```diff
@@ -0,0 +1,119 @@
+package metrics
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/ethereum-optimism/optimism/op-service/metrics"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+func keyLabels(labels []string) (common.Hash, error) {
+	sort.Strings(labels) // in-place sort strings so keys are order agnostic
+
+	encodedBytes, err := rlp.EncodeToBytes(labels)
+	if err != nil {
+		return common.Hash{}, err
+	}
+
+	hash := crypto.Keccak256Hash(encodedBytes)
+
+	return hash, nil
+}
+
+type MetricCountMap struct {
+	m *sync.Map
+}
+
+func NewCountMap() *MetricCountMap {
+	return &MetricCountMap{
+		m: new(sync.Map),
+	}
+}
+
+func (mcm *MetricCountMap) insert(values ...string) error {
+	key, err := keyLabels(values)
+
+	if err != nil {
+		return err
+	}
+
+	// update or add count entry
+	value, exists := mcm.m.Load(key.Hex())
+	if !exists {
+		mcm.m.Store(key.Hex(), uint64(1))
+		return nil
+	}
+	uint64Val, ok := value.(uint64)
+	if !ok {
+		return fmt.Errorf("could not read uint64 from sync map")
+	}
+
+	mcm.m.Store(key.Hex(), uint64Val+uint64(1))
+	return nil
+}
+
+func (mcm *MetricCountMap) Find(values ...string) (uint64, error) {
```

Comment:
Why `Find` and not `Get`, or `Read`? Find makes me think of finding the index, not the value.